api/api_test.go

```diff
@@ -164,3 +164,19 @@ func TestAPISignalAcceptsInt(t *testing.T) {
 
 // TODO
 func TestPersistence(t *testing.T) {}
+
+func TestConstructName(t *testing.T) {
+	signalName := "test_name"
+	r, _ := http.NewRequest("POST", "/ignored/anyway", nil)
+	r.RemoteAddr = "10.11.12.13:8089"
+
+	assert.Equal(t, constructName(signalName, r), "test_name@10.11.12.13")
+
+	r.Header = map[string][]string{"X-Forwarded-For": {"14.15.16.17"}}
```

Comment:
Please test X-Forwarded-For behavior in a different test. Ideally there should be 3 tests. One with no headers, and two others, each with one header. Maybe fourth with both headers.